pandas/tests/test_index.py

```diff
@@ -2815,6 +2815,13 @@ def testit():
             self.index.names = ["roger", "harold"]
         assertRaisesRegexp(ValueError, "^Length", testit)
 
+    def test_shift_index(self):
+        idx = Index(range(5))
```

Comment:
this should be in ``Base`` and use ``self.create_index()`` (override for the ``DatetimelikeIndexes`` to actually do a shift)